File: 01/main.go
package main

import (
	"fmt"
	"os"
)

func main() {
	b, err := os.ReadFile("input.txt")
	if err != nil {
		fmt.Print(err)
		return
	}
	str := string(b) // convert content to a 'string'

	floors := 0
	bii := 0
	for i, c := range str {
		if c == '(' {
			floors++
		}
		if c == ')' {
			floors--
		}
		if floors < 0 {
			if bii == 0 {
				bii = i + 1
			}
		}
	}
	fmt.Println("Part1: ", floors)
	fmt.Println("Part2: ", bii)
}

File: 01/main.js
const fs = require('node:fs');
fs.readFile('input.txt', 'utf8', (err, data) => {
    let floors = 0;
    let bii = 0;
    for (let i = 0; i < data.length; i++) {
        if (data[i] === "(" ) floors++;
        if (data[i] === ")" ) floors--;
        if (floors < 0) { if (!bii) bii = i + 1 }
    }
    console.log(`Part1: ${floors}`);
    console.log(`Part2: ${bii}`);
});